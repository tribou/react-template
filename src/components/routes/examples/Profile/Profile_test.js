// @flow

// Enzyme docs:
// http://airbnb.io/enzyme/docs/api/index.html

import React from "react";
import { shallow } from "enzyme";
import Profile from "./Profile";

type OverrideProps = {
  me: Object,
  error: string,
  location: Object,
  history: Object,
  fetchProfile: Function,
  token: ?string
};

const mockProps: OverrideProps = {
  me: {
    get: () => null
  },
  token: "test",
  error: "",
  location: {},
  history: {
    replace: () => {}
  },
  fetchProfile: () => {}
};

it("<Profile> renders the Profile content", () => {
  const wrapper = shallow(<Profile {...mockProps} />);

  expect(wrapper).toMatchSnapshot();
});
