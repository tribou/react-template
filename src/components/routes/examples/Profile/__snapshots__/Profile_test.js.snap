// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Profile> renders the Profile content 1`] = `
<div
  className="profile"
>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Profile"
  />
  <withRouter(Connect(RequireAuth)) />
  <div
    className="content"
  >
    Profile page
    <ul>
      <li />
      <li />
      <li />
    </ul>
    <Link
      replace={false}
      to="/home"
    >
      Back
    </Link>
    <button
      onClick={[Function]}
    >
      Refresh
    </button>
    <div
      className="error"
    />
  </div>
</div>
`;
